### PR TITLE
update doc and mergify rules v3.0.0 release

### PR DESCRIPTION
**Pull request by Madhu-1**
updated the upgrade, readme and mergify rules for v3.0.0 release

### CODE REVIEW

**Comment by Madhu-1** [comment]
@humblec @nixpanic can you please help me to merge this PR. I will branchout release-v3.0 once this PR is merged

File: .mergify.yml [hunk 1]
```diff
@@ -123,6 +123,30 @@ pull_request_rules:
...
+          - release-v3.0
+  # automerge backports if CI successfully ran
+  - name: automerge backport release-v3.0
+    conditions:
+      - author=mergify[bot]
+      - base=release-v3.0
+      - label!=DNM
+      - "#changes-requested-reviews-by=0"
+      - "#approved-reviews-by>=1"
+      - "status-success=continuous-integration/travis-ci/pr"
```
Comment (nixpanic) [inline .mergify.yml:142]: should `ci/centos/containerized-tests` not be required too?

Comment (Madhu-1) [reply .mergify.yml:142]: currently, it's not part of any release branch, we can do it later also after branching out. as Mergify works from the master branch

**Comment by Madhu-1** [comment]
Merging this one as the Mergify file is updated

**Comment by humblec** [comment]
Late to the party.. :). lgtm.

File: docs/ceph-csi-upgrade.md [hunk 2]
```diff
@@ -109,6 +110,12 @@ to upgrade from cephcsi 1.2 to v2.0
 
 ## Upgrading from v2.0 to v2.1
 
+Refer
+[upgrade-from-2.0-v2.1](https://github.com/ceph/ceph-csi/blob/v2.1.2/docs/ceph-csi-upgrade.md)
```
Comment (humblec) [inline docs/ceph-csi-upgrade.md:114]: @Madhu-1 just to mention it would have been better to `upgrade-from-v2.0-v2.1"

**Pull request merged**